internal/restorer/restorer_windows_test.go

```diff
@@ -4,12 +4,20 @@
 package restorer
 
```

Comment:
Added comprehensive testcase for backup and restore of all combinations of readonly, hidden, system, archive, encrypted and testcase for overriding of the attributes during restore.
Made other fixes as necessary, like fixes for restore of readonly files and encryption flag addition/removal as needed to fix the failing test cases.
Apologies for the size of the test case code.